### ISSUE TITLE
Remove machine upgrading

### ISSUE THREAD

**Issue by EmoGarbage404** [space-wizards/space-station-14#21055]
## Description

Based on a discord discussion.

Main points:
- numerical increases are fairly worthless and hard to make relevant to the station
- system has garbage integration with most machines and upgrades are worthless 99% of the time.
- it's a pain to implement and just adds a code burden.

The system as a whole is just pretty pointless. Upgrades are rarely meaningful outside of the ones strictly in science and the vertical progression of the part unlocks is just clunky as hell with research disciplines. 

just remove all the tech and parts. Feel free to maybe also buff some of the machines. If you need something for salv just replace it with tech gizmos or something you can sell.

**Comment by deltanedas** [comment]
only bit i agree with is that implementation is horrible

back when matter bins werent nerfed, you could upgrade things and actually save stupid mats which made it worth getting good parts, maybe just make number go up again

**Comment by metalgearsloth** [comment]
> only bit i agree with is that implementation is horrible
> 
> back when matter bins werent nerfed, you could upgrade things and actually save stupid mats which made it worth getting good parts, maybe just make number go up again

that's sorta the problem that only the lathe ones really matter and it's hard to integrate with anything else.

**Comment by EmoGarbage404** [comment]
does machine upgrading as whole offer any meaningful content outside of "make things in lathes cheaper"?

**Comment by Aisu9** [comment]
faster heating/cooking and reduce energy/biomass cost?
i agree we should see meaningful progress like more choice on the chem dispenser and other thing

**Comment by EmoGarbage404** [comment]
> faster heating/cooking and reduce energy/biomass cost? i agree we should see meaningful progress like more choice on the chem dispenser and other thing

biomass hasn't been relevant for months now and speed is pretty pointless when the scope of rounds is in hours. Meaningful progress can just be done with new items / machines in general which has the benefit of actually integrating with the research system as a whole. (adv mop, etc.)

**Comment by Aisu9** [comment]
i wonder why cloning is no longer relevant, can't a workaround be made to prevent "fuck it we clone"?

**Comment by deltanedas** [comment]
> that's sorta the problem that only the lathe ones really matter and it's hard to integrate with anything else.

so remove everything else 5% less power usage on a 400W machine doesnt matter

**Comment by metalgearsloth** [comment]
Is there a ss13 server that integrates upgrades well?

**Comment by EmoGarbage404** [comment]
> so remove everything else 5% less power usage on a 400W machine doesnt matter

then the discoverability and consistency becomes shitty as 90% of machines don't have any upgrades and whether or not a machine part does anything at all becomes a coin flip (funger reference).

**Comment by keronshb** [comment]
> Is there a ss13 server that integrates upgrades well?

Maybe Eris? They've had some interesting concepts.

**Comment by Ilya246** [comment]
what's the point of removing it? how does it make the game better?

**Comment by deltanedas** [comment]
machine upgrading is awful to implement

**Comment by Carolyn3114** [comment]
> what's the point of removing it? how does it make the game better?

ma bandaid fix!!

**Comment by mirrorcult** [comment]
removing things that arent good is okay

**Comment by Eden077** [comment]
Yeah I can't remember the last time it was worth wasting salvs and sciences time to come upgrade something

**Comment by Carolyn3114** [comment]
> removing things that arent good is okay

could perfectly just rebalance how a lot of upgrades work

**Comment by metalgearsloth** [comment]
balancing isn't the problem

**Comment by Carolyn3114** [comment]
> numerical increases are fairly worthless and hard to make relevant to the station

seems like it’s a major part in it

**Comment by Carolyn3114** [comment]
but oh well, half of the balancing here is just removing it

**Comment by LankLTE** [comment]
> does machine upgrading as whole offer any meaningful content outside of "make things in lathes cheaper"?

upgrading anomaly containers is arguably better and upgrading artifact scanners is very good too. Those are the ones I’d be really bummed to see go since they can be a big part of Science’s workflow.

**Comment by Carolyn3114** [comment]
upgrading SMES’s for more backup power, pacmans for more power efficiency, thermomachines for faster mixing, cloners for more efficiency, rechargers for faster recharge time, anomaly containers, lathes..

**Comment by Carolyn3114** [comment]
Plenty of good uses, just could use some rebalancing if it’s “not good enough” instead of just removing stuff

**Comment by Aisu9** [comment]
> upgrading SMES’s for more backup power, pacmans for more power efficiency, thermomachines for faster mixing, cloners for more efficiency, rechargers for faster recharge time, anomaly containers, lathes..

talking about the rechargers, know that the wall charger (that only charge weapons, not batteries) charge 5 whatever units faster than the desk one (from 20 to 25 base stats)

**Comment by metalgearsloth** [comment]
> upgrading SMES’s for more backup power, pacmans for more power efficiency, thermomachines for faster mixing, cloners for more efficiency, rechargers for faster recharge time, anomaly containers, lathes..

Power is plentiful on stations between solars / AME / TEG / Singulo so there's no reason to ever engage with power upgrades, and trying to micromanage balancing between mappers and code is just going to be unmaintainable.

Any other upgrades are accomplished by what emo already mentioned here:

> Meaningful progress can just be done with new items / machines in general which has the benefit of actually integrating with the research system as a whole. (adv mop, etc.)

It's not like "better lathe" is disappearing, just changing to something that will actually get used.

> Plenty of good uses, just could use some rebalancing if it’s “not good enough” instead of just removing stuff

No one has been doing it hence why the issue is made.

> it's a pain to implement and just adds a code burden.

**Comment by Carolyn3114** [comment]
I’m sure filling the entire gallery with a lot of “advanced (machine)” will solve the issue
advanced autolathe.. advanced protolathe.. advanced security techfab… advanced medical tech fab… advanced…….

**Comment by EmoGarbage404** [comment]
> I think machine upgrading is cool (like for the microwave or security charger or whatever else) I think the problem is balance. And if there's no reason to upgrade power THEN GIVE THEM BETTER UPGRADE TRAITS INSTEAD OF REMOVING A FEATURE THAT MULTIPLE PEOPLE ENJOY. 

If balance is so trivial, then just fix it yourself. And then also make sure that all future machines additionally implement upgrades in a meaningful way. And then make sure that the different components added to machine boards correspond to the machine in an easily understandable way.

...and even then, you still have a barely important feature that exists for little other than the personal betterment of science and *nothing else* with no incentive to improve anything for others.

> THEN GIVE THEM BETTER UPGRADE TRAITS INSTEAD OF REMOVING A FEATURE THAT MULTIPLE PEOPLE ENJOY. 

Tying in all caps doesn't make your point more meaningful it just makes you sound like a prick.

Machine upgrading is on a never-ending decline as it becomes less and less relevant outside of science boosting its own stuff and breaking balance.

**Comment by Ilya246** [comment]
i believe part upgrades are rarely used only because people forget that they exist and don't ask sci for them, for example thruster/gyro upgrades would be good for cargo yet cargo never asks for them, as well as medifab upgrades for med

**Comment by EmoGarbage404** [comment]
> i believe part upgrades are rarely used only because people forget that they exist and don't ask sci for them, for example thruster/gyro upgrades would be good for cargo yet cargo never asks for them, as well as medifab upgrades for med

If people genuinely forget they exist, then what benefit do they offer? Janitors don't forget that advanced mops exist. Yet machine upgrades are still good despite being irrelevant to 90% of players because ??? 

The mechanic as a whole is just an uninteresting maintenance burden. Anything it does could be done easier and better by just adding a new machine board and some sprites.

**Comment by Ilya246** [comment]
as for the burden on newly added machines, it's possible to just not add actual upgradeability to them (see: chemmaster, chem dispenser)
if this is done, there is literally no reason to remove it

**Comment by CatMagic1** [comment]
> as for the burden on newly added machines, it's possible to just not add actual upgradeability to them (see: chemmaster, chem dispenser) if this is done, there is literally no reason to remove it, as i believe lathe upgrades are balanced if someone doesn't like the machine not being upgradeable, they can PR to fix the issue

chem upgradeability will probably improve when chems inevitably become finite

**Comment by Cheackraze** [comment]
The problem isn't the system, the problem is that machines were never nerfed after upgrading system became a thing.  If machines were actually set to realistic values, upgrading everything would be much more worthwhile.

An actual finished implementation of the system should be attempted before its simply removed.

**Comment by metalgearsloth** [comment]
whoever is going to reduce the code maintenance and implement significantly more useful upgrades please make a draft PR.

**Comment by keronshb** [comment]
Citadel's SS13 server had upgrades from Tier 1 (base parts) - Tier 4 (Last Tier Parts)

Tier 1 machines & parts were dreadfully awful. Printing items from auto/proto-lathes could take upwards of 5+ seconds, and even extremely long times for the robotics machines. Tier 1 Batteries for typical devices like a baton or a disabler were actually fine.

Tier 2 and 3 parts were basically exclusively for hyper-minmaxing mining materials, since tier 2 and 3 parts gave miners more points & reclaimed more ore. Tier 2's mostly required just metal and glass &  Tier 3's required silver/uranium. A typical strat was to rush Tier 3's, have a miner insert a few uranium pieces, get the Tier 3 upgrades, then they'd dump a few diamond/gold pieces then push out Tier 4's for maximum points & ore reclamation.

Tier 1-3 mechs were completely useless, especially with batteries.

Tier 4 was the only thing ever worthwhile and that had given everything maximum efficiency, reclamation, etc. The only thing to outbeat tier 4 parts were yellow slime cores (self recharging cells).

We introduced an interesting mechanic on the stun batons, where a higher tier cell = better stunning, but it was only a small bump in the road since tier 4s were usually out sub 10-15 minutes (especially if you had people who can optimize mining & science).

Eventually we wanted to replace both techwebs & upgrading with something more meaningful but Cit died after awhile so no one really put the effort in.

One idea was to reduce the amount of parts to just cells, bins and processors. Maybe lasers was in there too? And to reduce the tiers from 1-4 to 1-2 or even 1-3. Part of this was to also cut down on the over-reliance on mining & ores. A 5th tier was talked about from unlocking alien tech but that was honestly power creep talk.

Each part would still have their own niche -
Cells - power consumption reduction
Bins - storage/reclamation 
Processors - speed

But that's just the same system with less part overhead.

While new machines can be added as an upgrade instead, we then have to deal with potential changing sprites and such.

I do like the idea of upgrades adding more functionality, but not at the cost of _removing_ functionality just to grant it back to the players.

Upgraded parts should exist to an extent for certain items.
Sidegraded parts/equipment should also exist

Better cells? Good for borgs/mechs/energy weps/chem/APCs/etc
Sidegrade cells? Maybe less capacity but they can self recharge.

**Comment by deltanedas** [comment]
> Cells - power consumption reduction

this is 100% useless since instead of getting a 50W saving on some random machine you could get a 5KW boost from a pacman if you are using that for some reason
if you are using an actual engine then the 50W saving doesnt matter anyway

**Comment by keronshb** [comment]
> > Cells - power consumption reduction
> 
> this is 100% useless since instead of getting a 50W saving on some random machine you could get a 5KW boost from a pacman if you are using that for some reason if you are using an actual engine then the 50W saving doesnt matter anyway

My brother in christ I was talking about specific scenarios 

- Energy guns
- Power limited based machines (Chem)
- Mechs
- Borgs
- etc

**Comment by deltanedas** [comment]
oh i thought you meant capacitors would they just be removed in a rework

**Comment by EmoGarbage404** [comment]
just to clarify: for this argument, cells are distinct from machine parts. Anything using power cells should just use the actual capacity of the cell to modify values, not the completely distinct machine part scaling system.

the only thing i'd maybe change about power cells is just making them all available roundstart and balancing with materials, but that's not here nor there.

**Comment by chromiumboy** [comment]
> just to clarify: for this argument, cells are distinct from machine parts. Anything using power cells should just use the actual capacity of the cell to modify values, not the completely distinct machine part scaling system.

All the power values would have to be rebalanced first. Not an easy task...

**Comment by BasedUser** [comment]
here's a list of machines where upgrades matter *right now*:
* lathes
* ore processors (also technically a lathe)
* thrusters
* thermomachines
* the artifact analyzer
* rechargers, to some extent
* SMES

where upgrades *could* matter, but don't have a profound enough effect:
- [ ] cloner
- [ ] chemmaster (increased pill size?)
- [ ] PACMANs (get LESS efficient with upgrades, not more)
- [ ] powered machines (too little power draw compensation)
- [ ] substations
- [ ] microwaves
- [ ] hotplates
- [ ] the ~~throngler~~ biomass reclaimer
- [ ] emitters

where upgrades should matter (not necessarily due to SS13), but can't (due to not having an upgrade slot or etc, some are for the best):
* APCs
* vending machines
* conveyors
* the recycler
* gas miners
* TEG
* airlocks (animation speed)

tldr this is stale just fix the broken upgrades instead of having an unmaintainability skill issue  
if i can read the part exchange code why can't you, the thruster system died before it
p.s. i may be portraying this as funnies but there are none to be found; go add docs or whatever it is maints do when not coding something actually worthwhile like richtext

**Comment by retequizzle** [comment]
> i may be portraying this as funnies but there are none to be found; go add docs or whatever it is maints do when not coding something actually worthwhile like richtext

If you cannot disagree with things being merged/discussed on Github without taking swipes at people directly, please fuck off. Like you said, there is nothing funny to be found here and the attempt to be a smarmy cunt does nothing to alleviate your concerns.

**Comment by BasedUser** [comment]
> re: retequizzle

I'm saying that if there isn't a coherent, planned out system that replaces machine upgrading (and none have been suggested here), don't bother.

**Comment by DangerRevolution** [comment]
Is the implementation bad?
Sure. Stats are arbitrary, and not widespread enough to have any significance. It's just something Sci decides to focus on some rounds and goes around focusing machines.
Is the idea bad? No.
This *could* add meaningful progression for machines,
+ something for Sci to focus on (I would actually argue this should be something for Engineering to do, throw them some sort of bone but that's not relevant)
+ Adds round progression.


> > re: @BasedUser
> 
> this issue doesn't really have like the full plan labeled out so i'm just gonna start over with what is good and bad and whatever.
> 
> So yeah, there are definitely upgrades and progression that are beneficial for the round and players. Lathes, artifact/anomaly equipment, and power infrastructure are all good candidates for improvement. I've always been a big candidate for internal round progression, so of course i think it's good to be able to make better machines for various purposes.
> 
> My main gripe, which this issue isn't very pointed in explaining, is that our conventional SS13 version of machine upgrading is kind of a terrible way of implementing that. Combined with the completely overhauled research system (NO MORE TECHWEB!!!!1!!!) it kinda feels like two incompatible systems in combat with each other. And, as a result, machine upgrading gets shoveled into this super weird spot where it's not really posed to be an impactful mechanic.
> 
> So i'm gonna just go through all the shit that makes Machine Upgrading a bad progression mechanic.
> 
> ### Research integration
> So at it's core, my biggest issue with machine upgrading is that it's a secondary progression mechanic that runs alongside tech disciplines while also being linked into them in the worst way possible.
> 
> The actual parts are shoveled in the experimental tree (at a thought, they should probably be in industrial?), but their application is for literally every discipline in the game. This partially contributes to experimental's dominance as a main discipline, since the machine parts exist as techs that are objectively good due to their incredibly wide application compared to literally anything else.
> 
> But this means that they are absolutely must-have's if you want to optimize your own or other departments. So in a way, it's basically a universal technology that must be shoved into some kind of discipline. This worked fine for the old techweb, but it runs pretty opposite to the intent of the new system, where rushing for the same unlock every round is not only wildly based on RNG but also defeating the purpose of variation between disciplines.
> 
> ### Consistency
> This is a point i've also elaborated on before, but it's worth re-iterating. For a lot of applications, parts are just an item you use for assembling a given machine. They aren't special and adding upgrades is somehow annoying enough that it just never gets done. It's less about current ones breaking, and more about future ones just never appearing. And while this isn't a technical degradation of any given upgrade, as a whole, they become less and less uniform as new things just stop supporting them.
> 
> I suppose you could just require every machine with parts to have upgrades coded, but you can't really enforce it with a test due to the nature of it and honestly i think people would just stop using them at all. Plus there's the fact that a lot of things just can't have sane upgrades with levels of variation that don't directly compete with just making more of them. As deltanedas stated, an upgrade for a microwave needs to be minimum twice as fast to be better than simply building another one to accomplish the same thing at scale.
> 
> ### Utility
> But even if you were to break down upgrades into only tier 1 and 4 (both ends of the extreme) I really don't see why getting a bunch of random machines to have upgrades is anything other than bloat. The reality is that a lot of equipment is just not that important to station function. If it's slow, people will just tolerate it and it being significantly faster doesn't matter all that much.
> 
> Even when things were made worse specifically to give room for upgrades, like reagent grinders, people just dealt with the changed values because number tweaks (which are basically the only thing this system can do sanely) are just not actually interesting enough to motivate players to engage with them beyond a spree of running around with an RPED. I am genuinely of the belief that, outside the few specific machines that people are regularly upgrading, there isn't a point to improving the vast majority of machines in the game (i mean, genuinely, who is the person upgrading the fat extractor)
> 
> And not to say that progression isn't important, but i just think all of these things could be made more useful, available, and easier to add via a different system, one that is in fact not formalized into any core thing.
> 
> My proposal is just to create new discrete machines that are simply improved versions of existing ones. Instead of upgrading an anomaly vessel, you unlock a new advanced anomaly vessel. Same for lathes and ore smelters and whatever else.
> 
> ### Research Integration
> In terms of research integration, it's a lot simpler. Relevant upgrades are now not only visible to players through the research UI, but are actually obtained through a progression that is relevant to what that upgrade does. If you want upgraded lathes that use fewer materials, then just make an efficient lathe in the Industrial discipline, where it actually belongs.
> 
> Along with this, you can now super easily implement new sound effects, visuals, and mechanics in an interesting way. Sure, you could do it before (technically, not that anything actually does) but now it can be done through YAML. And if you really want the experience of running around and upgrading en masse, you could make a machine that simply upgrades existing machine boards into their higher-research counterparts (i imagine that's what a modified RPED would do)
> 
> ### Consistency
> By freeing machine parts of any mechanical differences, they can just be a random filler piece for construction. Keeping things consistent isn't an issue because they no longer are struggling to operate on a shared format that doesn't benefit them. Upgrades can simply be their own things and be balanced on their own terms.
> 
> The majority of power-related upgrades can also be modeled simply with power cells and making the capacity of power cells (the only machine part with an additional use that integrates well with other mechanics) affect the capacity of the power infrastructure. This also means that they can just use construction for things like APCs which can't be modeled with existing machine construction.
> 
> ### Utility
> Now that upgrades can simply be created when they are actually necessitated by a feature, it's not really a question of "is upgrading this worth it?" All upgrades are implicitly applied to be valuable through their existence as unique things.
> 
> I think that by migrating the way we do machine upgrades to this methodology, not only can we create more upgrades that players are more likely to use and have fun with, but we can also make them so much deeper and more relevant to the game. They can actually be found easily and contribute to science instead of being the weird niche that they are now.
> 
> Upgrading moves from a set of parts to an intrinsic goal of the department.

I generally agree with this; there are issues but just ripping upgrades out instead of focusing on numerical changes is kind of counterproductive.

**Comment by Emisse** [comment]
maybe keep the (upgrade) but it just changes it into the new machine? with rped or something

**Comment by BasedUser** [comment]
> re: @EmoGarbage404

I see where this is coming from then, and now that it's more constructed, there are a few points I'd like to address that still irk me.

> But this means that they are absolutely must-have's if you want to optimize your own or other departments. \[\[Which means that it's heavily reliant on RNG and defeats the purpose of having a specialization to begin with.\]\]

Technically solvable by having two R&D servers instead of one... or \*shivers\* reverse engineering machines.  
By the way, I'm all for putting them in industrial instead. It'd even make more sense, since industrial is focused more on devices.

> ### Consistency

Actually, I'm leaning towards this point becoming *worse* after the removal of scanning modules and lasers, since there's less explicit directions an upgrade can have. But disregarding that, all of the "machine gets upgrades it can never use" instances can still be resolved, for example, making power use follow a `2^(init_rating-cumul_rating)` curve, and something analogous for other machines. A linear increase of +500% heat capacity in an atmos heater doesn't mean much when you could just build *more heaters*; or a SMES that stores only slightly more (due to how power ramping works by default, this doesn't help overall). These issues are genuinely atomic, and fixable given enough PRs, but there isn't a design doc on what the norm should be. The very idea of a design doc needs to be treated with care, since every Science and Engineering player has to remember if X machine's upgrade is multiplicative or linear. *Rest assured, it is doable* even in the new techweb.

> But even if you were to break down upgrades into only tier 1 and 4 (both ends of the extreme) I really don't see why getting a bunch of random machines to have upgrades is anything other than bloat. The reality is that a lot of equipment is just not that important to station function.

...Yet. Several machines are due for reworks, namely chem dispensers (no /tg/-like energy), thermomachines (along with the rest of atmos) and *maybe* the kitchen as a whole; and some equipment might benefit from upgrades where they never had any before. In my original comment, I mentioned upgrading airlocks. I wasn't exaggerating, if parts had the ability to be optional, this would be something a fork could implement on its own. It'd also open things like upgrading practice lasers *for shits and giggles* if that was ever coded in for guns.  
We don't have to go full Gregtech mode, but it could be possible to make T4s worth even for the average player.

> \[\[Instead of having parts, let's have advanced versions of boards that are specialized for *that exact thing.*\]\]

This is an honest attempt at decentralizing, but I don't think it's viable for the same reason that we brought up above: random machines getting upgrades. Why would Science, of all things, have to research an advanced microwave, if they could just *upgrade the ones they have* without spending extra research points on it? There's no need to do that, unless you're planning to give Salvage a field trip by searching for advanced versions of boards.

I might get axed for the following ideaguy suggestion:
* Why not just separate parts upgrading into a more expensive category of their own?

Maybe we're just approaching upgrades wrong. As far as I'm concerned, nobody's made a gray category yet. :godo:

<hr>

<sup> less formal answers ahead</sup>
> > maybe keep the (upgrade) but it just changes it into the new machine? with rped or something
>
> \[\[would rather have RPED upgrade the boards if they're researched\]\]

You could tie parts upgrading to boards, but that's... already been done by machine frames. I've no clue if it'd be useful or just code bloat. Also, you'd need to tie it to an R&D server.

**Comment by DangerRevolution** [comment]
From a discussion we've had downstream:

Maybe the move is to move upgrading into an Engineering thing, give them something to do? (See the end for why)

Fix up parts, maybe have a serious look at the research and parts acquisition issues but not just removing it outright.

*What each department is missing right now*

Sec:
Beepsky
Slavery
Station Criminal Records
Cool tech
Lots more antags

Med:
Surgery
Limb rework
Chemistry
Chemistry Factories
Virology

Sci:
More robotics
Publishing research
Xeno Biology
Maxcapping
Gas Mixing

Cargo:
Bit running
Lavaland
More wrecks & salvage
Mail rework
More cool tech & loot
Public Bounties
Private ordering

Engineering:
Tesla Crystal
Networking
And?
Why not parts upgrading?

**Comment by Hmeister-fake** [comment]
> From a discussion we've had downstream:
> 
> Maybe the move is to move upgrading into an Engineering thing, give them something to do? (See the end for why)
> 
> Fix up parts, maybe have a serious look at the research and parts acquisition issues but not just removing it outright.
> 
> _What each department is missing right now_
> 
> Sec: Beepsky Slavery Station Criminal Records Cool tech Lots more antags
> 
> Med: Surgery Limb rework Chemistry Chemistry Factories Virology
> 
> Sci: More robotics Publishing research Xeno Biology Maxcapping Gas Mixing
> 
> Cargo: Bit running Lavaland More wrecks & salvage Mail rework More cool tech & loot Public Bounties Private ordering
> 
> Engineering: Tesla Crystal Networking And? Why not parts upgrading?

When you coed it!!!!

Also 14 isn't 13, it's a successor. We don't need EVERY SINGLE THING copied from 13

**Comment by Hmeister-fake** [comment]
> When you coed it!!!!
> 
> Also 14 isn't 13, it's a successor. We don't need EVERY SINGLE THING copied from 13

Also even if it was copied over, a lot of things will be extremely unbalanced with stuff that is in 14. like the stun meta.

**Comment by Adrian16199** [comment]
I will be honest regardin this entire debate, parts are cool but alot more limited, regardin the machines, I think there should be only a: normal version and then upgraded version. not like 4 tiers of it. And the upgraded version has a really big meaningfull thing to it, like uh, an upgraded lathe I guess as an example: everythin costs half and hell, maybe works without power or some other gimmick.
somethin unique, make it cool, not just a 50% efficiency and a cool resprite with maybe a funny sound effect. make a microwave always cook in 5 seconds and let it choose what it can cook or suggest a recipe from anythin that you put it into. 
I do have to agree tho that engineers could use some more stuff to do. Not sure how that could be done but it would be nice.

**Comment by DangerRevolution** [comment]
> > From a discussion we've had downstream:
> > Maybe the move is to move upgrading into an Engineering thing, give them something to do? (See the end for why)
> > Fix up parts, maybe have a serious look at the research and parts acquisition issues but not just removing it outright.
> > _What each department is missing right now_
> > Sec: Beepsky Slavery Station Criminal Records Cool tech Lots more antags
> > Med: Surgery Limb rework Chemistry Chemistry Factories Virology
> > Sci: More robotics Publishing research Xeno Biology Maxcapping Gas Mixing
> > Cargo: Bit running Lavaland More wrecks & salvage Mail rework More cool tech & loot Public Bounties Private ordering
> > Engineering: Tesla Crystal Networking And? Why not parts upgrading?
> 
> When you coed it!!!!
> 
> Also 14 isn't 13, it's a successor. We don't need EVERY SINGLE THING copied from 13

I am aware it's a "wyci"; however my point was that Engineering was lacking, is lacking and will be lacking and this could be something cool to rework and give to Engineering...

**Comment by EmoGarbage404** [comment]
this is veering kinda off-topic but most of my points have been made so /shrug

how would machine upgrading even work as an engineering thing. parts have always been tied to RND as a system and i don't really see a system that would let them be obtained by engineers in a logical way. They hardly even fabricate things in scale.

**Comment by IamVelcroboy** [comment]
> how would machine upgrading even work as an engineering thing. parts have always been tied to RND as a system and i don't really see a system that would let them be obtained by engineers in a logical way. They hardly even fabricate things in scale.

RnD researches the tech that unlocks the upgrades. Engineering are the tinkerers that do the actual upgrading. Or, could/should be.

**Comment by DangerRevolution** [comment]
> > how would machine upgrading even work as an engineering thing. parts have always been tied to RND as a system and i don't really see a system that would let them be obtained by engineers in a logical way. They hardly even fabricate things in scale.
> 
> RnD researches the tech that unlocks the upgrades. Engineering are the tinkerers that do the actual upgrading. Or, could/should be.

This ^

**Comment by EmoGarbage404** [comment]
i'm gonna rewrite all this shit into a doc because i took a fat public relations L on explaining most of my ideas.

**Comment by EmoGarbage404** [comment]
https://github.com/space-wizards/docs/pull/119

proper doc and discussion should go here.
